### PR TITLE
Adding queuing delay to latency analysis for async buses

### LINKED ISSUE #1148: Missing queuing latency handling for buses
We currently deal with sampling latency for periodic protocols and buses.
We could do queuing latency based on the number of connections bound to the bus

### PR DESCRIPTION
**Pull request by AaronGreenhouse**
Fixes #1148

### CODE REVIEW

File: analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java [hunk 1]
```diff
@@ -835,41 +974,6 @@ public AnalysisResult invoke(ComponentInstance ci, SystemOperationMode som, bool
 		return results;
 	}
 
-	public AnalysisResult invoke(EndToEndFlowInstance etef) {
```
Comment (lwrage) [inline analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java]: Why did you remove this method? It could be used by a 3rd party plugin.

File: analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java [hunk 2]
```diff
@@ -835,41 +974,6 @@ public AnalysisResult invoke(ComponentInstance ci, SystemOperationMode som, bool
...
-	 *
-	 * @param etef The end to end flow instance
-	 * @param som The mode to run the analysis in. If null then run all SOMs
-	 * @param asynchronousSystem Whether the system is treated as asynchronous
-	 * @param majorFrameDelay Whether partition output is performed at a major frame (as opposed to the partition end)
-	 * @param worstCaseDeadline Use deadline based processing (as opposed to max compute execution time)
-	 * @param bestCaseEmptyQueue Assume empty queue (instead of full)
-	 * @return A populated report in AnalysisResult format.
-	 */
-	public AnalysisResult invoke(EndToEndFlowInstance etef, SystemOperationMode som, boolean asynchronousSystem,
```
Comment (lwrage) [inline analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java]: Why did you remove this method? It could be used by a 3rd party plugin.

File: analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java [hunk 3]
```diff
@@ -788,10 +921,10 @@ public AnalysisResult invoke(ComponentInstance ci) {
 	 * @param bestCaseEmptyQueue Assume empty queue (instead of full)
 	 * @return A populated report in AnalysisResult format.
 	 */
-	public AnalysisResult invoke(ComponentInstance ci, SystemOperationMode som, boolean asynchronousSystem,
+	private AnalysisResult invoke(ComponentInstance ci, SystemOperationMode som, boolean asynchronousSystem,
```
Comment (lwrage) [inline analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java]: "invoke" methods are meant to be used by other plugins so they should be public.

File: analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java (hunk 1 above)
Comment (AaronGreenhouse) [reply analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java]: I put it back

File: analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java (hunk 2 above)
Comment (AaronGreenhouse) [reply analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/FlowLatencyAnalysisSwitch.java]: I put it back

**Comment by AaronGreenhouse** [comment]
I put back the invoke() methods.  I got carried away yesterday...

**Comment by lwrage** [comment]
@reteprelief Could you check if the change does what it should do, please?
